File: Solutions/01-99/11. Container With Most Water/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = -1
        
        left, right = 0, len(height)-1

        while left < right:
            l = min(height[left], height[right])
            b = right - left

            if max_area < l*b:
                max_area = l*b
            
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return max_area

File: Solutions/01-99/11. Container With Most Water/Solution.cpp
class Solution 
{
public:
    int maxArea(vector<int>& height) 
    {
        int max_area = 0, left = 0, right = height.size()-1;

        while(left < right)
        {
            int l = min(height[left], height[right]);
            int b = right - left;
            max_area = max(max_area, l*b);

            if(height[left] < height[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }

        return max_area;
    }
};
